toolkit/analysis_and_visualization/classification/assets/scripts/app.js

```diff
@@ -0,0 +1,572 @@
+}
+
+// eslint-disable-next-line no-unused-vars
+function highlightRow(element) {
+    var parentRow = element.parentElement.parentElement;
+    parentRow.classList.toggle("highlight-row");
+
+}
+
+
+//Create copy of imageSummary
+var imageSummaryFiltered = JSON.parse(JSON.stringify(imageSummary));
+
+// eslint-disable-next-line no-unused-vars
+function clearResultFilter() {
+    $("input[id ^= \"fli_\"]").val("");
+    imageSummaryFiltered = JSON.parse(JSON.stringify(imageSummary));
+    loadImageResults(imageSummaryFiltered);
+    updateItems();
+}
+
+function filterResultTable(e) {
+    imageSummaryFiltered = [];
+    var allEmpty = true;
+    if (e) {
+
+        // eslint-disable-next-line no-unused-vars
+        var code = (e.keyCode ? e.keyCode : e.which);
+    }
+    var combinationOp = $("input:radio[name='filterType']:checked").val();
+    var notOp = $("#not-op").is(":checked");
+
+
+    var filters = {};
+
+    // eslint-disable-next-line no-unused-vars
+    $("input[id ^= \"fli_\"]").each(function (element) {
+        var allNames = this.id.split("_");
+        var property_index = null;
+        var prop_name_new = allNames[1];
+        if (allNames[2]) {
+            property_index = parseInt(allNames[2], 10);
+            prop_name_new = prop_name_new + "_" + property_index;
+        }
+        var value = this.value;
+        var op = $("#fli_op_" + prop_name_new).val();
+        if (value) {
+            allEmpty = false;
+            filters[this.id] = [];
+            filters[this.id].push(op);
+            filters[this.id].push(value);
+
+        }
+    });
+
+    imageSummaryFiltered = imageSummary.filter(function (item) {
+        var allTrue = true;
+        var anyTrue = false;
+
+        for (var key in filters) {
+            var allNames = key.split("_");
+            var property = allNames[1];
+            var property_index = null;
+            if (allNames[2]) {
+                property_index = parseInt(allNames[2], 10);
+            }
+            var itemValue = item[property];
+            if (allNames[2] >= 0) {
+                itemValue = itemValue[property_index];
+            }
+
+            if (!compare(itemValue, filters[key][1], filters[key][0])) {
+                allTrue = false;
+            } else {
+                anyTrue = true;
+            }
+
+
+        }
+        if (notOp === true) {
+            anyTrue = !anyTrue;
+            allTrue = !allTrue;
+        }
+        if (combinationOp == "or") {
+            return anyTrue;
+        } else if (combinationOp == "and") {
+            return allTrue;
+        }
+
+        return true;
+
+    });
+
+    if (allEmpty) {
+        var perPage = 50;
+        var pageNumber = 1;
+        imageSummaryFiltered = JSON.parse(JSON.stringify(imageSummary));
+        var showFrom = perPage * (pageNumber - 1);
+        var showTo = showFrom + perPage;
+        loadImageResults(imageSummaryFiltered.slice(showFrom, showTo));
```

Comment:
Codacy found an issue: ['loadImageResults' was used before it was defined.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6397592)